maro/simulator/scenarios/oncall_routing/business_engine.py

```diff
@@ -55,12 +55,23 @@ def _get_staying_time(tick: int, plan: List[PlanElement], order_transition: Dott
 
     cur_tick = tick
     processing_time = 0
+    processing_order = False
     for order in copied_orders:
         order_status = order.get_status(cur_tick, order_transition)
-        if order_status != OrderStatus.DUMMY:
-            if order_status == OrderStatus.NOT_READY:
-                cur_tick = order.open_time - order_transition.buffer_before_open_time
-                processing_time = 0  # TODO
+        if order_status in (OrderStatus.DUMMY, OrderStatus.COMPLETED, OrderStatus.TERMINATED):
+            pass
+        elif order_status == OrderStatus.NOT_READY:
+            if processing_order and not order_transition.processing_proportion_to_quantity:
+                processing_time = order_transition.processing_time
+
+            cur_tick = max(
+                order.open_time - order_transition.buffer_before_open_time,
+                cur_tick + processing_time
+            )
+            processing_time = 0
+            processing_order = False
+        else:
+            processing_order = True
             if order_transition.processing_proportion_to_quantity:
                 processing_time += order_transition.processing_time
 
```

Comment:
if processing_order and not order_transition.processing_proportion_to_quantity:
                processing_time = order_transition.processing_time